### DISCUSSION TITLE
Cookie Documentation inconsistent with server-only module

### DISCUSSION THREAD

**Discussion by dvnrsn** [remix-run/remix#9900] [General]
Given that
> The build will fail if any code in a .server file or .server directory accidentally ends up in the client module graph.

https://remix.run/docs/en/main/discussion/server-vs-client#server-modules

It seems an adjustment is needed to the [cookies documentation](https://remix.run/docs/en/main/utils/cookies#using-cookies) where cookie parsing functions are exported from `.server` files and used in actions resulting in

```
[commonjs--resolver] Server-only module referenced by client
```

I suppose one could dynamically import the function. But is it a problem if this is exposed to client? In other words, should this file not be of extension `.server` in the docs?

**Comment by dvnrsn** [comment]
I failed to see `   Remix automatically removes server-code from these exports:
    loader, action, headers`
    
There was a floating function in my file that was referencing cookies

**Reply by sergiodxa** [reply]
If you called createCookie inside you route, because createCookie could cause a side-effect (the compiler doesn't know) then even if nothing uses the result it will keep it just in case. You need to call createCookie in a server-only file or inside loader/action.
